skiko/src/awtMain/objectiveC/macos/MetalOffScreenContextHandler.mm

```diff
@@ -0,0 +1,45 @@
+#ifdef SK_METAL
+
+#import <jawt.h>
+#import <jawt_md.h>
+
+#import <QuartzCore/CAMetalLayer.h>
+#import <Metal/Metal.h>
+#import <GrDirectContext.h>
+#import <mtl/GrMtlBackendContext.h>
+#import <mtl/GrMtlTypes.h>
+
+#import "MetalDevice.h"
+
+extern "C"
+{
+
+JNIEXPORT jlong JNICALL Java_org_jetbrains_skiko_context_MetalOffScreenContextHandler_makeMetalContext(
+        JNIEnv *env, jobject contextHandler, jlong adapterPtr) {
+    @autoreleasepool {
+        id <MTLDevice> adapter = (__bridge id <MTLDevice>) (void *) adapterPtr;
+        GrMtlBackendContext backendContext = {};
+        backendContext.fDevice.retain((__bridge GrMTLHandle) adapter);
+        id <MTLCommandQueue> fQueue = [adapter newCommandQueue];
+        backendContext.fQueue.retain((__bridge GrMTLHandle) fQueue);
+        return (jlong) GrDirectContext::MakeMetal(backendContext).release();
+    }
+}
+
+JNIEXPORT jlong JNICALL Java_org_jetbrains_skiko_context_MetalOffScreenContextHandler_makeMetalRenderTargetOffScreen(
+        JNIEnv *env, jobject contextHandler, jlong adapterPtr, jint width, jint height) {
+    @autoreleasepool {
+        id <MTLDevice> adapter = (__bridge id <MTLDevice>) (void *) adapterPtr;
+        MTLTextureDescriptor *textureDescriptor = [MTLTextureDescriptor texture2DDescriptorWithPixelFormat:MTLPixelFormatBGRA8Unorm width:width height:height mipmapped:NO];
+        // TODO: use double buffer
+        id <MTLTexture> metalTexture = [adapter newTextureWithDescriptor:textureDescriptor];
+        GrMtlTextureInfo info;
+        info.fTexture.retain((__bridge GrMTLHandle) metalTexture);
+        GrBackendRenderTarget *renderTarget = NULL;
+        renderTarget = new GrBackendRenderTarget(width, height, 0, info);
```

Comment:
Same concern about memory leak. Should we add finalizer?

Reply:
As I noticed from previous PR, it will be cleared by Skiko, isn't?

Reply:
We not just found object with the same type. I read the code and traced the path of this pointer. Probably one more finalizer might lead to use after free or double freeing

Reply:
Okay. It should be passed to `BackendRenderTarget` to be deleted. It's true in this case, but I don't really like such memory management.

Possible solution: multiple constructors or factory methods in the same object to handle all memory management in the same object. Such solution allows to limit possible wrong usage of API in the future